.github/workflows/codeql-analysis.yml

```diff
@@ -4,8 +4,15 @@ on:
     branches: [ main ]
   pull_request:
 
+permissions:
+  contents: read
+
 jobs:
   CodeQL-Build:
+    permissions:
+      actions: read  # for github/codeql-action/init to get workflow details
```

Comment:
Is this not by default?

Reply:
No, it isn't. But that is required for Private repos. So I think for this one it isn't required.